SQL/0000-00-00-schema.sql

```diff
@@ -1643,6 +1671,386 @@ CREATE TABLE `parameter_type_override` (
+  `SpecimenTypeID` integer unsigned NOT NULL,                                   
+  `ContainerTypeID` integer unsigned NOT NULL,                                  
+  `Regex` varchar(255) NOT NULL,                                                
+  CONSTRAINT `PK_biobank_validate_identifer` PRIMARY KEY (SpecimenTypeID, ContainerTypeID),
+  CONSTRAINT `FK_biobank_validate_identifier_SpecimenTypeID`                    
+    FOREIGN KEY (`SpecimenTypeID`) REFERENCES `biobank_specimen_type`(`SpecimenTypeID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,                                      
+  CONSTRAINT `FK_biobank_validate_identifier_ContainerTypeID`                   
+    FOREIGN KEY (`ContainerTypeID`) REFERENCES `biobank_container_type`(`ContainerTypeID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT                                       
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;                                        
+                                                                                
+CREATE TABLE `biobank_specimen_type_unit_rel` (                                 
+  `SpecimenTypeID` integer unsigned NOT NULL,                                   
+  `UnitID` integer unsigned  NOT NULL,                                          
+  CONSTRAINT `PK_biobank_container_psc_rel` PRIMARY KEY (`SpecimenTypeID`, `UnitID`),
+  CONSTRAINT `FK_biobank_specimen_type_unit_rel_TypeID`                         
+    FOREIGN KEY (`SpecimenTypeID`) REFERENCES `biobank_specimen_type` (`SpecimenTypeID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,                                      
+  CONSTRAINT `FK_biobank_specimen_type_unit_rel_SourceID`                       
+    FOREIGN KEY (`UnitID`) REFERENCES `biobank_unit` (`UnitID`)                 
+    ON UPDATE RESTRICT ON DELETE RESTRICT                                       
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;                                        
+                                                                                
+CREATE TABLE `biobank_specimen_parent` (                                        
+  `SpecimenID` integer unsigned NOT NULL,                                       
+  `ParentSpecimenID` integer unsigned NOT NULL,                                 
+  CONSTRAINT `PK_biobank_specimen_parent` PRIMARY KEY (`SpecimenID`, `ParentSpecimenID`),
+  CONSTRAINT `FK_biobank_specimen_parent_SpecimenID`                            
+    FOREIGN KEY (`SpecimenID`) REFERENCES `biobank_specimen`(`SpecimenID`)      
+    ON UPDATE RESTRICT ON DELETE RESTRICT,                                      
+  CONSTRAINT `FK_biobank_specimen_parent_ParentSpecimenID`                      
+    FOREIGN KEY (`ParentSpecimenID`) REFERENCES `biobank_specimen`(`SpecimenID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT                                       
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;                                        
```

Comment:
A specimen can have multiple parent and "children". Right?

Reply:
Yes